crypto-common/src/lib.rs

```diff
@@ -258,6 +275,11 @@ pub trait KeyIvInit: KeySizeUser + IvSizeUser + Sized {
         let iv = Self::generate_iv_with_rng(rng)?;
         Ok((key, iv))
     }
+
+    /// Check if an IV might be considered weak
+    fn weak_iv_test(_iv: &Iv<Self>) -> Result<(), WeakKeyError> {
+        Ok(())
+    }
```

Comment:
I don't think we need  this method. Plus, we can add later it in a backward-compatible way if it's needed for something after all.

Reply:
Can I ask you take over?
I'm not sure I can follow what you're asking or how you want the traits to be laid out.

Reply:
Ok, I can do it a bit later. Though this PR is almost ready in my opinion. You just need to remove `KeyIvInit::weak_iv_test` and keep `KeyIvInit::weak_key_test`.

Reply:
You also could allow edits by maintainers (there should be a checkbox on the right) and I can make the necessary edits.

Reply:
Restored the branch, you now have access (somehow I thought it was on by default? but it appears not)

I didn't mean to be rude, I'm just not sure I can follow the entire logic of the Inner trait and how they are meant to integrate together, and I thought that would have been easier for you to just write the code.

Reply:
No problem. I already created #1742 based on this PR.